tests/src/core/CachedPackageLocator.js

```diff
@@ -0,0 +1,212 @@
+import fs from 'fs'
+import sinon from 'sinon'
+import { expect } from 'chai'
+
+import CachedPackageLocator from '../../../src/core/CachedPackageLocator'
+
+function notEmpty(obj) {
+  return Object.keys(obj).length
+}
+
+function reduce({
+  dependencies = {},
+  devDependencies = {},
+  peerDependencies = {},
+  optionalDependencies = {},
+} = {}) {
+  if ([dependencies, devDependencies, peerDependencies, optionalDependencies].some(notEmpty)) {
+    return { dependencies, devDependencies, peerDependencies, optionalDependencies }
+  }
+
+  return null
+}
+
+describe('CachedPackageLocator.readUpSync()', function () {
+  let sandbox
+  let packageLocator
+
+  const withUnexpectedTokenStr = '{{"name":"foo"}'
+  const withUnexpectedEndStr = '{"name":"foo"'
+  const withDeps = {
+    dependencies: { 'prop-types': '~15.0.0' },
+    devDependencies: { 'webpack': '^2.0.0' },
+    peerDependencies: { 'react': '>=15.0.0' },
+    optionalDependencies: { 'fs-events': '*' },
+  }
+  const withDepsStr = JSON.stringify(withDeps)
+  const withDepsExtraFieldsStr = JSON.stringify(
+    Object.assign({}, withDeps, {
+      name: 'not-needed',
+      description: 'foo bar',
+    })
+  )
+  const context = {
+    report: sinon.spy(),
+  }
+
+  before(function () {
+    sandbox = sinon.sandbox.create()
+    sandbox.stub(fs, 'readFileSync').reset()
+  })
+
+  after(function () {
+    sandbox.restore()
```

Comment:
this should be done `afterEach`.

Reply:
I understand, that's what I had originally also - it does not initialize the stubbed method without calling reset, though. Try it yourself!

Reply:
I'll play with it; but in the meantime let's change all this stuff so the sandbox is created `before` but everything else is in a beforeEach/afterEach.